pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -728,6 +746,19 @@ func (vs *VSphere) DiskIsAttached(volPath string, nodeName k8stypes.NodeName) (b
 	}
 	requestTime := time.Now()
 	isAttached, err := diskIsAttachedInternal(volPath, nodeName)
+	if err != nil {
+		if IsManagedObjectNotFoundError(err) {
+			glog.V(4).Infof("error %q ManagedObjectNotFound for node %q", err, convertToString(nodeName))
+			err = vs.nodeManager.RediscoverNode(nodeName)
+			if err != nil {
+				if IsManagedObjectNotFoundError(err) {
```

Comment:
I did not understand this logic, could you explain why we return isAttached as false and err as nil in this case?

Reply:
I do not see any testing done related to the changes in DiskIsAttached/DIsksAreAttached. How did you test this code?